src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/WindowsRuntime/Windows/UI/ViewManagement/InputPane.cs

```diff
@@ -62,21 +57,26 @@ internal class InputPane : IDisposable
             [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
             static InputPane()
             {
+                // WinRT types are only available on Win10+
+                if (Environment.OSVersion.Platform == PlatformID.Win32NT && Environment.OSVersion.Version < new Version(10, 0))
```

Comment:
Can it check the win10 system version?

> For applications that have been manifested for Windows 8.1 or Windows 10. Applications not manifested for Windows 8.1 or Windows 10 will return the Windows 8 OS version value (6.2). To manifest your applications for Windows 8.1 or Windows 10, refer to Targeting your application for Windows.

See https://docs.microsoft.com/en-us/windows/win32/sysinfo/operating-system-version?wt.mc_id=MVP

Reply:
Yes, `Environment.OSVersion` was recently updated to get the real OS version, not the one calculated based on the manifest.

Reply:
In WPF code base, we do more precise version checks that overcomes appcompat manifest lies. Environment.OSVersion should not be used. 

Use this instead https://github.com/dotnet/wpf/blob/8c7ab1592b6c70e732106aeabf58f215d4b71c22/src/Microsoft.DotNet.Wpf/src/Shared/System/Windows/InterOp/OSVersionHelper.cs

Reply:
It seems to me that the EntryPointNotFoundException should be what determines support (a capability check) and not an OS version check.  Is there a reason why we absolutely require a version check here?

Reply:
Mainly for performance reasons since exceptions are slow. I can remove the OS checks and we can use exceptions to detect lack of support.

Reply:
I don't believe there is much of a perf impact here.  This constructor is only running on demand the first time we attempt to deal with the InputPane (which [TipTsfHelper](https://github.com/dotnet/wpf/blob/master/src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/Interop/TipTsfHelper.cs) only does when necessary).  After that we know the answer to whether or not the platform is supported and future calls are basically noops or useful work.

I'd prefer we stick to the capability check (since this OS check is actually not specific enough anyway, it needs to ensure 1607+).